### PR TITLE
Adjust to new auoms input format

### PR DESCRIPTION
**Pull request by taglines**

### CODE REVIEW

**Comment by lagalbra** [comment]
@Microsoft/omsagent-devs

File: source/code/plugins/in_auoms.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+require 'cool.io'
+require 'yajl'
+
+require 'fluent/input'
+require 'fluent/event'
+
+module Fluent
+  # obsolete
+  class AuOMSInput < Input
+    Plugin.register_input('auoms', self)
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: Please follow our naming convention: in_auoms

File: source/code/plugins/in_auoms.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
+#
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: Where possible, unit tests should be added for this plugin.

File: source/code/plugins/in_auoms.rb [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+require 'socket'
+
+require 'cool.io'
+require 'yajl'
+
+require 'fluent/input'
+require 'fluent/event'
+
+module Fluent
+  # obsolete
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: What's obsolete?

File: source/code/plugins/in_auoms.rb [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    desc 'Tag of the output events.'
+    config_param :tag, :string, default: nil
+    desc 'The path to your Unix Domain Socket.'
+    config_param :path, :string, default: nil
+    desc 'The backlog of Unix Domain Socket.'
+    config_param :backlog, :integer, default: nil
+
+    def initialize
+      require 'socket'
+      require 'yajl'
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: You already require yajl and socket at the top of the file; why do you need them here?

File: source/code/plugins/in_auoms.rb [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      @lsock.close
+      @thread.join
+    end
+
+    def listen
+      if File.exist?(@path)
+        File.unlink(@path)
+      end
+      FileUtils.mkdir_p File.dirname(@path)
+      log.info "listening fluent socket on #{@path}"
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: Use $log.info "message"

Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: nit: change to "Listening to FluentD socket on #{@path}"

File: source/code/plugins/in_auoms.rb [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+      s = Coolio::UNIXServer.new(@path, Handler, log, method(:on_message))
+      s.listen(@backlog) unless @backlog.nil?
+      s
+    end
+
+    def run
+      @loop.run(@blocking_timeout)
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: You're missing a `begin` keyword here.

File: source/code/plugins/in_auoms.rb [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    rescue
+      log.error "unexpected error", error: $!.to_s
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: For all logging, use $log.info and $log.error

File: source/code/plugins/in_auoms.rb [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        end
+        @on_message = on_message
+        @log = log
+        @log.trace {
+          remote_port, remote_addr = *Socket.unpack_sockaddr_in(@_io.getpeername) rescue nil
+          "accepted fluent socket from '#{remote_addr}:#{remote_port}': object_id=#{self.object_id}"
+        }
+      end
+
+      def on_connect
```
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: Why is this empty method here?

File: source/code/plugins/in_auoms.rb (hunk 2 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: Done

File: source/code/plugins/in_auoms.rb (hunk 3 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: Removed. Was cruft from in_stream.rb

File: source/code/plugins/in_auoms.rb (hunk 1 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: This is the correct naming convention. ALL in_<name>.rb input plugins use: Plugin.register_input('<name>', self)

File: source/code/plugins/in_auoms.rb (hunk 4 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: Removed.

File: source/code/plugins/in_auoms.rb (hunk 5 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: Changed to @log

File: source/code/plugins/in_auoms.rb (hunk 6 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: This is valid. If the "rescue" is at the same level as "def' then begin isn't required.

File: source/code/plugins/in_auoms.rb (hunk 7 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: Changed to @log

File: source/code/plugins/in_auoms.rb (hunk 8 above)
Comment (taglines) [inline source/code/plugins/in_auoms.rb]: Not empty anymore.

File: source/code/plugins/in_auoms.rb (hunk 6 above)
Comment (lagalbra) [inline source/code/plugins/in_auoms.rb]: Huh, interesting!

**Comment by lagalbra** [comment]
Please confirm that you have run pBuild with these changes, as well as onboarding the OMSAgent with these changes and verifying the impacted scenarios.

I'm unfamiliar with the purpose of these plugins/what they log, so it might be good from that perspective to get a reviewer from your team.

**Comment by taglines** [comment]
Yes, I've run pBuild. Yes, I've onboarded (many times), and tested upgrade scenario.

**Pull request merged**